app/models/concerns/orchestration/compute.rb

```diff
@@ -290,9 +290,13 @@ def match_macs_to_nics(fog_attr)
     # array so we can delete from it safely
     fog_nics = vm.interfaces.dup
 
+    logger.debug "Orchestration::Compute: Trying to match network interfaces from fog #{fog_nics.inspect}"
     self.interfaces.each do |nic|
       selected_nic = vm.select_nic(fog_nics, nic)
-      next if selected_nic.nil? # found no matching fog nic for this Foreman nic, move on
+      if selected_nic.nil? # found no matching fog nic for this Foreman nic
+        logger.warn "Orchestration::Compute: Could not match network interface #{nic.inspect}"
+        return failure(_("Could not find virtual machine network interface matching %s") % nic.identifier || nic.ip || nic.name || nic.type)
```

Comment:
I think you may need some brackets around the `||` part, as precedence means it may not work properly:

    irb(main):001:0> "%s" % nil || :a
    => ""